securedrop/journalist.py

```diff
@@ -359,19 +346,53 @@ def admin_delete_user(user_id):
     return redirect(url_for('admin_index'))
 
 
-@app.route('/account', methods=('GET', 'POST'))
+@app.route('/account', methods=('GET',))
 @login_required
 def edit_account():
-    if request.method == 'POST':
+    password = make_password()
+    return render_template('edit_account.html',
+                           password=password)
+
+
+@app.route('/account/new-password', methods=['POST'])
+@login_required
+def new_password():
+    user = g.user
+    password = request.form.get('password')
+    _new_password(user, password)
+    return redirect(url_for('edit_account'))
+
+
+def make_password():
+    while True:
+        password = crypto_util.genrandomid(7)
         try:
-            edit_account_password(g.user, request.form['password'],
-                                  request.form['password_again'])
-        except (PasswordMismatchError, InvalidPasswordLength):
-            return redirect(url_for('edit_account'))
+            Journalist.check_password_acceptable(password)
+            return password
+        except PasswordError:  # pragma: no cover
+            continue
+
 
-        commit_account_changes(g.user)
+def _new_password(user, password):
+    try:
+        user.set_password(password)
+    except PasswordError:
+        flash('You submitted a bad password! Password not changed.', 'error')
+        return
 
-    return render_template('edit_account.html')
+    try:
+        db_session.commit()
+    except Exception:  # pragma: no cover
```

Comment:
Hmm: is there a specific exception you are intending to catch here? Can we mock `db_session.commit()` to add a unit test instead of no cover?